glue/utils/tests/test_array.py

```diff
@@ -114,3 +116,35 @@ def test_unbroadcast():
     z = unbroadcast(y)
     assert z.shape == (1, 1, 3)
     np.testing.assert_allclose(z[0, 0], x)
+
+
+@pytest.mark.parametrize(('chunk_shape', 'n_max'),
+                         [(None, 121), (None, 100000), (None, 5), ((3, 1, 2, 4, 1, 2), None)])
+def test_iterate_chunks(chunk_shape, n_max):
+
+    array = np.zeros((6, 3, 4, 5, 1, 8))
+
+    for slices in iterate_chunks(array.shape, chunk_shape=chunk_shape, n_max=n_max):
+
+        print(slices)
+
+        # Make sure empty slices aren't returned
+        assert array[slices].size > 0
+
+        # Increment all values in slice by 1
+        array[slices] += 1
+
+    assert_equal(array, 1)
+
+
+def test_iterate_chunks_invalid():
+
+    array = np.zeros((6, 3, 4, 5, 1, 8))
+
+    with pytest.raises(ValueError) as exc:
+        next(iterate_chunks(array.shape, chunk_shape=(6, 2, 1)))
+    assert exc.value.args[0] == 'chunk_shape should have the same length as shape'
```

Comment:
See https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions
version 3 has the message argument

Reply:
Good to know! I might just change it everywhere in a separate PR, will open an issue.